### ISSUE TITLE
HiDPI plugin issues

### ISSUE THREAD

**Issue by hfiguiere** [muse-sequencer/muse#804]
This is a followup to #697 / #710 

tl;dr it still isn't working right.

I run on a HiDPI laptop. This is 3.1

With two different version of Dexed **VST** (Linux, build using default)

**Revert Native GUI Scaling is OFF (default)**

Dexed VST from DISTRHO-Ports (doesn't support HiDPI):

![Screenshot from 2020-05-15 19-31-29](https://user-images.githubusercontent.com/114441/82103905-d9425600-96e2-11ea-9782-6bde55e5adbe.png)

Dexed VST 0.9.4h from upstream (do support HiDPI):

![Screenshot from 2020-05-15 19-31-41](https://user-images.githubusercontent.com/114441/82103917-de9fa080-96e2-11ea-9009-84c89f60ed40.png)

**Revert Native GUI Scaling is ON**

Dexed VST from DISTRHO-Ports:

![Screenshot from 2020-05-15 19-34-35](https://user-images.githubusercontent.com/114441/82103991-2b837700-96e3-11ea-9a60-06651e5a265f.png)

Dexed VST 0.9.4h from upstream:

![Screenshot from 2020-05-15 19-34-42](https://user-images.githubusercontent.com/114441/82103994-2f16fe00-96e3-11ea-885f-0f8167825921.png)

While I'm not sure what can be done, one option I could see is being able to flip the option per plugin, but something that just work is better.. 

Also is it feasible to be able to zoom? I'd love to be able to say "I want that plugin zoomed 200%" as it is otherwise unreable. (VST or LV2)

Let me know if you need more information.

**Comment by terminator356** [comment]
> one option I could see is being able to flip the option per plugin

Hi. It might be a question for @kybos since I don't have a high dpi screen.
He has been doing most of the work in this area.
But I thought I would throw this in:
If you open the generic UI for the plugin (not the native UI), there is a
 **per-plugin instance** "Revert native UI HiDpi scaling" override button
 found on the UI's 'quirks' toolbar.
Might that help?

**Comment by kybos** [comment]
I wrote a wiki recently on this topic:

https://github.com/muse-sequencer/muse/wiki/HiDPI
(So as Tim wrote above, there is a per plugin setting to solve exactly this problem.)

There is no solution that "just works", and there will never be, given the nature of the issue. The native plugin GUIs are not controllable from outside (they are separate apps in this respect, mostly even using a completely different UI technology), so the only thing a DAW can try is to set the initial window size to fit the GUI.

**Comment by hfiguiere** [comment]
That setting is well hidden when you don't know it is there. :-/ So this works.

**Comment by hfiguiere** [comment]
and indeed Surge (LV2) isn't fixable.

And as for the override button, I can say that the tri state button isn't the most usable. Even the colour difference isn't very clear: I get confused myself. I would have done a popup menu.

**Comment by StaffanMelin** [comment]
@hfiguiere Surge has its own menu (bottom right) with a zoom sub-menu! Very useful! :)

**Comment by hfiguiere** [comment]
It is so small that it is hard to see the small menu button down there. 

If I soom to 300% it tells me that 295% is the max because my screen size (ok I'll take that), there are still white borders around: if I zoom, the window size increase.
Also if I close the window and reopen, surge ui size is reset *inside* but not the window size after zooming, and it seems I can't reach the menu button again. Apparenlty I missed I can set it to default in surge though.

(this is surge LV2 1.6.6 built from upstream)

**Comment by hfiguiere** [comment]
> And as for the override button, I can say that the tri state button isn't the most usable. Even the colour difference isn't very clear: I get confused myself. I would have done a popup menu.

Just to clarify, I have to use the tooltip to understand the values, and the default is "use global setting" but it doesn't even tell me what the global setting is so I have to guess.

**Comment by StaffanMelin** [comment]
> It is so small that it is hard to see the small menu button down there.

Well, I don't really understand your description, sorry. But for me, Surge now always opens at 200% and is working 100%! :)

My screen is 15.6 inch at 1920x1280 pixels. I had a lot of hidipi issues before (both with QTractor and MusE) but now, for me, MusE handles it without problems.

I run on Debian Buster with XFCE4, if that matters...

**Comment by hfiguiere** [comment]
> My screen is 15.6 inch at 1920x1280 pixels. I had a lot of hidipi issues before (both with QTractor and MusE) but now, for me, MusE handles it without problems.

This is not HiDPI. 

3200x1800 on a 13" is. So at 100% Surge is barely readable.  Hence I didn't notice it.

And yes the preference is saved, but still the window is improperly sized.

FYI here is how it looks by default (screen truncated, MuSE UI as size reference):

![Screenshot from 2020-05-17 14-56-53](https://user-images.githubusercontent.com/114441/82157318-c2b90d80-984e-11ea-9cf8-9693b1e39748.png)

The "menu" button isn't more than 1mm tall.

**Comment by StaffanMelin** [comment]
I see. Looks troublesome...sorry.
